src/XamlIl.Cecil/CecilMethod.cs

```diff
@@ -86,7 +96,17 @@ class CecilMethod : CecilMethodBase, IXamlIlMethodBuilder
                 && DeclaringType.Equals(cm.DeclaringType)
                 && Reference.FullName == cm.Reference.FullName;
 
+            public IXamlIlMethod MakeGenericMethod(IReadOnlyList<IXamlIlType> typeArguments)
```

Comment:
I think we need a way to check if method is a generic method definition and make [FindMethod overloads](https://github.com/kekekeks/XamlIl/blob/master/src/XamlIl/TypeSystem/TypeSystem.cs#L338) to check for that. Probably only the one with `FindMethodMethodSignature` should return generic definitions if allowed by said `FindMethodMethodSignature`. Not exactly a requirement for merging, but would be nice to have.

Reply:
Also, check if parameters list is correct for constructed generic methods. Might be worth adding a test for that.